wfe2/wfe_test.go

```diff
@@ -267,10 +264,11 @@ func makeBody(s string) io.ReadCloser {
 	return ioutil.NopCloser(strings.NewReader(s))
 }
 
-// loadPrivateKey loads a private key from PEM/DER-encoded data.
-// Duplicates functionality from jose v1's util.LoadPrivateKey function. It was
+// loadKeys loads a private key from PEM/DER-encoded data and returns both the
+// private key and corresponding public key.
+// Originally duplicated from from jose v1's util.LoadPrivateKey function. It was
 // moved to the jose-util cmd's main packge in v2.
-func loadPrivateKey(t *testing.T, keyBytes []byte) interface{} {
+func loadKeys(t *testing.T, keyBytes []byte) (interface{}, interface{}) {
```

Comment:
I think rather than refactoring this to return the public key, you can define it to return `crypto.Signer`, which defines a `.Public()` method returning `crypto.PublicKey`. Callers can then call `.Public()` to get the public key. This has the advantage that it returns a `*rsa.PublicKey` or `*ecdsa.PublicKey` rather than the `rsa.PublicKey` or `ecdsa.PublicKey` that you are currently returning. This follows the more customary (and efficient) pattern of using pointers for keys, and I think it will allow you to remove the special test-only `sigAlgForKey` in favor of the more general version that expectes pointers.